compiler/src/dotty/tools/dotc/typer/Inferencing.scala

```diff
@@ -626,11 +626,13 @@ object Inferencing {
     case tp: RecType => tp.derivedRecType(captureWildcards(tp.parent))
     case tp: LazyRef => captureWildcards(tp.ref)
     case tp: AnnotatedType => tp.derivedAnnotatedType(captureWildcards(tp.parent), tp.annot)
+    case tp: FlexibleType => tp.derivedFlexibleType(captureWildcards(tp.hi))
     case _ => tp
   }
 
   def hasCaptureConversionArg(tp: Type)(using Context): Boolean = tp match
     case tp: AppliedType => tp.args.exists(_.typeSymbol == defn.TypeBox_CAP)
+    case tp: FlexibleType => hasCaptureConversionArg(tp.hi)
```

Comment:
For this one, I don't understand why the other cases are not needed (i.e. what makes `FlexibleType` special). For example, what if you get the right `AppliedType` under an `AnnontatedType`? In that case, this method won't find it either.

But the case for `FlexibleType` looks right to me.

Reply:
I guess `hasCaptureConversionArg` is only used to check top-level `AppliedType` of a tree for error messages. Maybe adding a `stripped` after `tp` can handle more cases, idk. Given the places where `FlexibleType` is expected to appear, I just added this case.